enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -761,6 +761,11 @@ func (mc *MigrationCache) copyDataInBackground() error {
 
 		select {
 		case <-mc.quitChan:
+			// Drain copy channel on shutdown
+			for len(mc.copyChan) > 0 {
```

Comment:
can you leave a short comment why we're draining the channel in this manner. it's always good to leave a comment when doing something slightly unusual.